guides/source/active_record_basics.md

```diff
@@ -220,134 +313,205 @@ The application will have to use the [`id_value`][] alias attribute to access th
+book.inspect
+# => "#<Book id: 106, title: \"The Lord of the Rings\", author: \"J.R.R. Tolkien\", created_at: \"2024-03-04 19:15:58.033967000 +0000\", updated_at: \"2024-03-04 19:15:58.033967000 +0000\">"
 ```
 
-Using the `new` method, an object can be instantiated without being saved:
+While the `new` method will instantiate an object *without* saving it to the database:
 
 ```ruby
-user = User.new
-user.name = "David"
-user.occupation = "Code Artist"
-```
+book = Book.new
+book.title = "The Hobbit"
+book.author = "J.R.R. Tolkien"
 
-A call to `user.save` will commit the record to the database.
+# Note that the `id` is not set for this object.
+book.inspect
+# => "#<Book id: nil, title: \"The Hobbit\", author: \"J.R.R. Tolkien\", created_at: nil, updated_at: nil>"
+
+# The above `book` is not yet saved to the database.
+
+book.save
+book.id # => 107
+
+# Now the `book` record is committed to the database and has an `id`.
+```
 
 Finally, if a block is provided, both `create` and `new` will yield the new
 object to that block for initialization, while only `create` will persist
 the resulting object to the database:
 
 ```ruby
-user = User.new do |u|
-  u.name = "David"
-  u.occupation = "Code Artist"
+book = Book.new do |b|
+  b.title = "Metaprogramming Ruby 2"
+  b.author = "Paolo Perrotta"
 end
+
+book.save
+```
+
+The resulting SQL statement from both `book.save` and `Book.create` look something like this:
+
+```sql
+/* Note that `created_at` and `updated_at` are automatically set */
+
+INSERT INTO "books" ("title", "author", "created_at", "updated_at") VALUES (?, ?, ?, ?) RETURNING "id"  [["title", "Metaprogramming Ruby 2"], ["author", "Paolo Perrotta"], ["created_at", "2024-02-22 20:01:18.469952"], ["updated_at", "2024-02-22 20:01:18.469952"]]
 ```
 
 ### Read
 
-Active Record provides a rich API for accessing data within a database. Below
-are a few examples of different data access methods provided by Active Record.
+Active Record provides a rich API for accessing data within a database. You can query a single record or multiple records, filter them by any attribute, order them, group them, select specific fields, and do anything you can do with SQL.
 
 ```ruby
-# return a collection with all users
-users = User.all
+# Return a collection with all books
+books = Book.all
+
+# Return a single book
+first_book = Book.first
+last_book = Book.last
+book = Book.take
 ```
 
-```ruby
-# return the first user
-user = User.first
+The above results in the following SQL:
+
+```sql
+# Book.all
+SELECT "books".* FROM "books"
+
+# Book.first
+SELECT "books".* FROM "books" ORDER BY "books"."id" ASC LIMIT ?  [["LIMIT", 1]]
+
+# Book.last
+SELECT "books".* FROM "books" ORDER BY "books"."id" DESC LIMIT ?  [["LIMIT", 1]]
+
+# Book.take
+SELECT "books".* FROM "books" LIMIT ?  [["LIMIT", 1]]
 ```
 
+We can also find specific books with `find_by` and `where`. While `find_by` returns a single record, `where` returns an array:
```

Comment:
I don't know if eventually someone will come by saying it's technically not "an array" until it's loaded/accessed, etc. (probably not, but I could see it happening...), maybe we can sidestep that by saying it returns "a list of records"?

```suggestion
We can also find specific books with `find_by` and `where`. While `find_by` returns a single record, `where` returns a list of records:
```